src/ui/PipelineContext/PipelineContext.ts

```diff
@@ -90,24 +90,36 @@ export class PipelineContext extends Component {
    * @param key
    * @returns {string}
    */
-  public getContextValue(key: string): string {
-    return this.content[key].replace(/\{\!([^\}]+)\}/g, (all: string, contextKey: string) => {
-      if (Coveo.context != null && contextKey in Coveo.context) {
-        return Coveo.context[contextKey];
-      } else if (contextKey == PipelineContext.CURRENT_URL) {
-        return window.location.href;
-      }
-      return '';
-    });
+  public getContextValue(key: string): string | string[] {
+    if (_.isArray(this.content[key])) {
+      let contextValues = [];
+      _.each(this.content[key], (value) => {
+        contextValues.push(this.getModifiedData(value));
+      });
+      return contextValues;
+    } else {
+      return this.getModifiedData(this.content[key]);
+    }
   }
 
   private handleBuildingQuery(args: IBuildingQueryEventArgs) {
     let keys = this.getContextKeys();
     _.each(keys, (key: string) => {
       args.queryBuilder.addContextValue(key, this.getContextValue(key));
+
     });
   }
 
+  private getModifiedData(value: string) {
```

Comment:
It was (and still is) a bit obscure for someone new that would come into the codebase why we need to "modify" the data.

I would add some comments here to explain that's it's used to escape specials salesforce characters. eg:  `{!   }`